docs/articles/announcements/Towards-NUnit4.md

```diff
@@ -0,0 +1,201 @@
+  
+### Developer details
+
+The improved result messages have been implemented using the new [CallerArgumentExpression](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/caller-argument-expression) together with using a [FormattableString](https://learn.microsoft.com/en-us/dotnet/api/system.formattablestring?view=net-6.0) as the class for the message.  
+
+## Proper async/await, you can now await the Asserts
+
+NUnit 3 has implemented async functionality using a pattern called 'sync-over-async.' This approach allowed for its implementation without requiring significant underlying changes.
+
+In version 4 proper async/await has been implemented with a series of new assert methods, `Assert.ThatAsync` which can be `await`'ed.
+
+```cs
+    [Test]
+    public async Task AssertionPasses_CompletedTaskWithResult_EqualsResult()
+    {
+        await Assert.ThatAsync(() => Task.FromResult(42), Is.EqualTo(42));
+    }
+```
+
+Version 4 also introduces a new feature called Assert.MultipleAsync, which allows you to mix async and sync asserts within the same block. This enables you to perform multiple assertions, both asynchronous and synchronous, in a more concise and streamlined manner. Moreover, it's important to note that Assert.MultipleAsync is awaitable, providing flexibility in handling asynchronous operations and assertions.
+
+```cs
+    [Test]
+    public async Task AssertMultipleAsyncSucceeds()
+    {
+        await Assert.MultipleAsync(async () =>
+        {
+            await Assert.ThatAsync(() => Task.FromResult(42), Is.EqualTo(42));
+            Assert.That("hello", Is.EqualTo("hello"));
+            await Assert.ThatAsync(() => Task.FromException(new  ArgumentNullException)), Throws.ArgumentNullException);
+        });
+    }
+```
+
+## Nullability
+
+Version 4 has implemented stricter nullability throughout the codebase. While there are still some areas that require fixing, overall, it should now conform to the nullability requirements.  See [3376](https://github.com/nunit/nunit/issues/3376) for details.
+
+These changes, along with the improved assert messages, have resulted in null values no longer being allowed for messages. As a consequence, code like the example below will not compile and will result in [CS0121](https://learn.microsoft.com/en-us/dotnet/csharp/misc/cs0121?f1url=%3FappId%3Droslyn%26k%3Dk(CS0121))):
+
+![AssertNull](~/images/AssertNull.png)
+It's important to update such code to use valid non-null message strings
+
+## Platform support
+
+The lowest framework platforms support in Version 4 are **.net framework 4.6.2** and **.net 6.0**.
+
+## Classic/Legacy asserts
+
+The classic/legacy asserts, like `Assert.AreEqual`, have now been moved into its own project, and will be released as a separate package.  They are now in the namespace `NUnit.Framework.Classic`.
+In the early V4 alpha version they will be delivered in the standard NUnit package.
```

Comment:
⚠️ We should probably make it clear, then, that we would expect that to be a future breaking change, which means end users wouldn't be affected until v5 in the new release structure. Or are we saying that it will be a breaking change between v4 alpha and v4 release?

Reply:
It will be breaking in V4, since you need to change the namespace to be able to use the legacy asserts.
Adding the package we can then do as part of the V4, but not for the first alpha release